File: benchmarks/call-graph-accuracy/fixtures/python/registry.py
# Call-graph accuracy fixture (Python).
# Patterns:
#   - direct call:    parse_line() inside setup()
#   - decorator:      @log_calls
#   - function reference (v1.11.1+): register("evt", on_event), map(parse_line, ...)


def log_calls(fn):
    def wrap(*a, **kw):
        return fn(*a, **kw)

    return wrap


def parse_line(line):
    return line.strip()


def on_event(payload):
    return payload


@log_calls
def setup():
    register("evt", on_event)
    pipe = list(map(parse_line, ["a", "b"]))
    parse_line("seed")
    return pipe

File: benchmarks/call-graph-accuracy/fixtures/rust/registry.rs
// Call-graph accuracy fixture (Rust).
// Patterns:
//   - direct call: handler() inside run()
//   - method chain: builder.with_x().with_y()
//   - macro:        my_log!()
//   - function reference (v1.11.0+): LazyLock::new(build_tools), iter.map(parse_line)

fn build_tools() -> Vec<u32> {
    vec![1, 2, 3]
}

fn parse_line(s: &str) -> u32 {
    s.len() as u32
}

fn handler() {
    let _ = build_tools();
}

macro_rules! my_log {
    ($($t:tt)*) => {};
}

static TOOLS: std::sync::LazyLock<Vec<u32>> = std::sync::LazyLock::new(build_tools);

fn run() {
    handler();
    my_log!("hi");
    let lines = ["a", "bb"];
    let _: Vec<_> = lines.iter().map(parse_line).collect();
}
